PlasmaMVP/feeStructure.md

```diff
@@ -1,17 +1,38 @@
-* In the case of a valid fee withdrawal, his exit will not be challenged and his fee will be sent after the 1 week period.
-* In the case of an invalid fee withdrawal, users will see that and initiate a mass withdrawal. Since the validator has the lowest priority, all other users' exits will be processed before his, leaving him with the true fee amount.
 
 ### Fees Stored in Utxos
-Fees are included in the transaction. When a validator creates a block, he reserves a utxo at the last leaf (to give himself the lowest priority in the block) of the block where he can aggregate all the fees of this block. This utxo will look like a deposit utxo. Users can check that the amount stored in this fee utxo is correct. If the amount or transaction is found to be invalid, users will mass exit.
+In our earlier discussion of a plasma side chain vulnerability, [Ethereum Research Post](https://ethresear.ch/t/plasma-vulnerabiltity-sybil-txs-drained-contract/1654), regarding the plasma mvp spec not specifying the requirement of storing confirm sigs on-chain, we can exploit this to solve our issues with fees!  
+
+By requiring correct confirm signatures be included in a transaction, spending a utxo on the child chain allows anyone watching to successfully challenge an invalid attempt to exit any ancestor of that utxo. We claim that that this requirement makes it very unlikely to exit a invalid uxto successfully.  
+
+We can take advantage of this by having a validator collect fees corresponding to the **inputs** of all utxos in his/her block. As specified in the txBytes, the fees must be deducted from the inputs. A valid spend message on the child chain follows this equality:  
+
+    Amount1 + Amount2 - Fee = Output1 + Output2  
+
+The fees are deducted from the first input and determined by the participants and not validators. This allows validators to choose which spend messages to include from the mempool. However due to the high volume, the market will most likely settle at an average low transaction fee  
+
+While processing each tx, the validator aggregates all input fees, and creates a new utxo with no history owned by them as the very last utxo in the block. By placing this new utxo last, all watchers of this child chain can verify the validity of the denomination of the new utxo by aggregating the fees for each tx themselves. Malicious activity of this last tx is grounds for a mass exit of the child chain. Placing the new utxo last also ensures that the validator has the lowest priority to exit with that block as the current head.  
+
+There are two different cases in which we assure that the validator is garunteed their fee utxo for a given block.
+1. For every transaction included in a block, any spend of the new utxos allows the validator to challenge an invalid exit that would steal the validator fees
```

Comment:
I don't understand this case. Elaborate?

Reply:
Once a user spends their transaction and broadcast their confirm signatures, they theoretically are not suppose to withdraw the previous utxo. Withdrawing this utxo is not only invalid, but would steal the fees of the validator who included the transaction of that spend message. The act of the new parent utxo spending their output allows the validator to challenge using the confirm sig